src/config.c

```diff
@@ -682,14 +840,32 @@ int ReadConfig(RedisModuleString **argv, int argc, char **err) {
       return REDISMODULE_ERR;
     }
 
-    // `triggerId` is set by the coordinator when it registers a trigger for a configuration.
-    // If we don't have a coordinator or this configuration has no trigger, this value
-    // is meaningless and should be ignored
-    if (curVar->setValue(&RSGlobalConfig, &ac, curVar->triggerId, &status) != REDISMODULE_OK) {
-      *err = rm_strdup(QueryError_GetError(&status));
-      QueryError_ClearError(&status);
-      return REDISMODULE_ERR;
+    // if the value is different from default, we can't update it, because the
+    // CONFIG SET has higher priority than the FT.CONFIG SET
+    RSConfig RSDefaultConfig = RS_DEFAULT_CONFIG;
+    sds currValue = curVar->getValue(&RSGlobalConfig);
+    sds currValueDefault = curVar->getValue(&RSDefaultConfig);
+
+    if ((currValue != NULL && currValueDefault!= NULL
```

Comment:
can you store the condition in a boolean variable name called `equals`?
IMHO it would simplify the reading of the code.

Reply:
Maybe even put the error case(inequality) at the start of the function to simplify the code.